openmicroscopy/bioformats
# Split Bio-Formats C++ from bioformats.git

- Drop moved files (bioformats, qtwidgets, docs, examples)
- Install missing ome-xml header; it was private but needs to be public now it's separate
- Register ome-xml paths (schemas, transforms, samples); samples are used to find test samples in dependent modules
- Provide model version information in public header; needed so that dependent modules know which model version they are using for transforms etc.
- Documentation uses doxygen tagfiles to generate cross-references to one-common

The new bioformats-cpp component will build using https://github.com/rleigh-dundee/bioformats-cpp/commits/develop (the CI jobs have been updated to clone this until it's adopted by the ome organisation).

Some additional work will be needed with the one-common path registration and lookup code to allow it to support multiple search locations.  It currently works for the purposes of making the split work, but isn't going to work in all circumstances.

--------

Testing:

- Builds should remain green
- Documentation should have correct cross-references between the bioformats and ome-xml and one-common docs, and between the ome-xml and ome-common docs
- Check that `bf-test info` still works using a superbuild binary archive; this will verify that the path introspection still works
- Check that the contents of the super-build and regular build binary archives are unchanged (bar the fixes to the api reference paths).  A diff of the file listings should be sufficient to check this.
